src/modules/service.js

```diff
@@ -1,94 +1,62 @@
-    return new Promise((fulfill, reject) => {
-      // wild    no load no save
-      // dryrun  load not save
-      // record  load and save
-      let options = this._options(endpoint);
+  async get(endpoint) {
+    // wild    no load no save
+    // dryrun  load not save
+    // record  load and save
+    const options = this.options(endpoint);
+    const { cacheMode } = this.config;
+    let response;
 
-      if (this.config.cacheMode === 'wild') {
-        this.log.debug(`wild -- ${options.uri}`);
-        request.get(options, (err, response, body) => {
-          if (!err) {
-            if (response.statusCode === 404) {
-              // special case for 404 because the Student Web Service
-              // returns ugly HTML in the response body.
-              body = 'Not found.';
-            }
-            fulfill(this._buildResult(response, body));
-          } else {
-            reject(err);
-          }
-        });
-      } else if (this.config.cacheMode === 'dryrun') {
-        this.log.debug(`dryrun for ${options.uri}`);
-        let body = this.cache.read(options.uriCache);
-        if (body) {
-          let response = {};
-          response.statusCode = 200;
-          fulfill(this._buildResult(response, body));
-        } else {
-          request.get(options, (err, response, body) => {
-            if (!err) {
-              if (response.statusCode === 404) {
-                // special case for 404 because the Student Web Service
-                // returns ugly HTML in the response body.
-                body = 'Not found.';
-              }
-              fulfill(this._buildResult(response, body));
-            } else {
-              reject(err);
-            }
-          });
-        }
-      } else if (this.config.cacheMode === 'record') {
-        this.log.debug(`record -- ${options.uri}`);
-        let body = this.cache.read(options.uriCache);
-        if (body) {
-          let response = {};
-          response.statusCode = 200;
-          fulfill(this._buildResult(response, body));
-        } else {
-          request.get(options, (err, response, body) => {
-            if (!err) {
-              if (response.statusCode === 200) {
-                this.cache.write(options.uriCache, body, true);
-              } else if (response.statusCode === 404) {
-                // special case for 404 because the Student Web Service
-                // returns ugly HTML in the response body.
-                body = 'Not found.';
-              }
-              fulfill(this._buildResult(response, body));
-            } else {
-              reject(err);
-            }
-          });
-        }
+    if (cacheMode !== 'wild') {
+      const body = this.cache.read(options.uriCache);
+
+      if (body) {
+        response.statusCode = 200;
+        response.body = body;
+        this.log.debug(`${cacheMode} cache hit for ${options.url}`);
+      } else {
+        this.log.debug(`${cacheMode} cache miss for ${options.url}`);
```

Comment:
Are we just logging it when cache is missed?

Reply:
Yes. Need to know where the data is coming from if you're debugging.